donut/auth_utils.py

```diff
@@ -359,9 +368,31 @@ def is_caltech_user():
         re.search(ip_match, flask.request.remote_addr) is not None
 
 
-def is_admin():
+def is_admin(user_id=None):
```

Comment:
If `user_id` is passed in, we should be able to pass it directly to `check_permission()` (and `get_permissions()` as well). Currently, this code looks up the username from the user ID and then looks up the user ID from the username. Ideally we should find all calls to `check_permission()` where the user ID could be passed in directly (to avoid a DB lookup).